File: tutorial/code/python/switch-notify.py
import zio

def input_changed(event): # <1>
    global switch_pin

    if event.pin != switch_pin: # <2>
        return
    if event.value:
        print "Switch Off"
    else:
        print "Switch On"

agent = zio.Agent("/dev/ttyUSB0")
gpio = zio.GPIO(agent)
switch_pin = 0
gpio.add_change_handler(input_changed) # <3>

agent.wait_for_events() # <4>

File: tutorial/code/java/SwitchNotify.java
import com.zilogic.zio.*;

class SwitchNotify {
    public static void main(String args[]) 
	throws ProtocolException, InterruptedException {

	Agent agent = new Agent("/dev/ttyUSB0");
	GPIO gpio = new GPIO(agent);

	GPIOChangeListener listener = new GPIOChangeListener() {
		public void inputChanged(GPIOChangeEvent event) { // <1>
		    int switchPin = 0;
		    
		    if (event.getPin() != switchPin) // <2>
			return;

		    if (event.getValue() == 1)
			System.out.println("Switch Off");
		    else
			System.out.println("Switch On");
		}
	    };

	gpio.addChangeListener(listener); // <3>
	agent.waitForEvents(); // <4>
    }
}
